js/signup.js

```diff
@@ -0,0 +1,73 @@
+let signupEmail = document.querySelector('#signup-email');
+let signupNickname = document.querySelector('#signup-nickname');
+let signupPassword = document.querySelector('#signup-password');
+let signupCheckPassword = document.querySelector('#signup-check-password');
+let signupBtn = document.querySelector('.signup-btn');
+
+
+function inputSignupEmail(){
+  activeSignupBtn()
+}
+
+function inputSignupNickname(){
+  activeSignupBtn()
+}
+
+function inputSignupPassword(){
+  activeSignupBtn()
+}
+
+function inputSignupCheckPassword(){
+  activeSignupBtn()
+}
+
+function activeSignupBtn(){
+  if( signupEmail.value.length > 0 &&
+      signupNickname.value.length > 0 &&
+      signupPassword.value.length > 0 &&
+      signupCheckPassword.value.length > 0 &&
+      signupPassword.value === signupCheckPassword.value
+  ) {
+    signupBtn.classList.add('active');
+  }
+}
+
+
+signupEmail.addEventListener('input', inputSignupEmail);
+signupNickname.addEventListener('input', inputSignupNickname);
+signupPassword.addEventListener('input', inputSignupPassword);
+signupCheckPassword.addEventListener('input', inputSignupCheckPassword);
+//signupBtn.addEventListener('click')
+
+
+let SignupPasswordVisibility = document.querySelector('.signup-password-visibility');
+let SignupCheckpasswordVisibility = document.querySelector('.signup-checkpassword-visibility');
+
+console.log(SignupPasswordVisibility);
+
+
+function setSignupPasswordVisibility() {
+  let img = SignupPasswordVisibility.getAttribute('src');
+  if(img === './img/btn_visibility_off.svg'){
+    SignupPasswordVisibility.setAttribute('src', './img/btn_visibility_on.svg');
+    SignupPasswordVisibility.previousElementSibling.setAttribute('type', 'text');
+  } else {
+    SignupPasswordVisibility.setAttribute('src', './img/btn_visibility_off.svg');
+    SignupPasswordVisibility.previousElementSibling.setAttribute('type', 'password');
+  }
+}
+
+function setSignupCheckpasswordVisibility() {
+  let img = SignupCheckpasswordVisibility.getAttribute('src');
+  if(img === './img/btn_visibility_off.svg'){
+    SignupCheckpasswordVisibility.setAttribute('src', './img/btn_visibility_on.svg');
+    SignupCheckpasswordVisibility.previousElementSibling.setAttribute('type', 'text');
+  } else {
+    SignupCheckpasswordVisibility.setAttribute('src', './img/btn_visibility_off.svg');
+    SignupCheckpasswordVisibility.previousElementSibling.setAttribute('type', 'password');
+  }
+}
+
+
+SignupPasswordVisibility.addEventListener('click', setSignupPasswordVisibility);
+SignupCheckpasswordVisibility.addEventListener('click', setSignupCheckpasswordVisibility);
```

Comment:
두개 비슷한 함수가 있는데, 아래처럼 하나로 통합할 수 있어요.
```suggestion
function togglePasswordVisibility(visibilityToggle) {
  const img = visibilityToggle.getAttribute('src');
  const newImg = img === './img/btn_visibility_off.svg' ? './img/btn_visibility_on.svg' : './img/btn_visibility_off.svg';
  const inputType = newImg.includes('on') ? 'text' : 'password';

  visibilityToggle.setAttribute('src', newImg);
  visibilityToggle.previousElementSibling.setAttribute('type', inputType);
}

[signupEmail, signupNickname, signupPassword, signupCheckPassword].forEach(input => {
  input.addEventListener('input', activeSignupBtn);
});

signupPasswordVisibility.addEventListener('click', () => togglePasswordVisibility(signupPasswordVisibility));
signupCheckPasswordVisibility.addEventListener('click', () => togglePasswordVisibility(signupCheckPasswordVisibility));
```

Reply:
변수명은 SignupPasswordVisibility -> 이거 같은 경우 PascalCase 인데, 여기서는 camelCase를 사용하는게 맞습니다.
```
signupPasswordVisibility
```